src/Build/Logging/BinaryLogger/BinaryLogger.cs

```diff
@@ -88,10 +89,11 @@ public enum ProjectImportsCollectionMode
         public void Initialize(IEventSource eventSource)
         {
             _initialTargetOutputLogging = Environment.GetEnvironmentVariable("MSBUILDTARGETOUTPUTLOGGING");
-            _initialLogImports = Environment.GetEnvironmentVariable("MSBUILDLOGIMPORTS");
+            _initialLogImports = Traits.Instance.EscapeHatches.LogProjectImports;
 
             Environment.SetEnvironmentVariable("MSBUILDTARGETOUTPUTLOGGING", "true");
             Environment.SetEnvironmentVariable("MSBUILDLOGIMPORTS", "1");
```

Comment:
we probably don't need this, and the logger should check the Escape hatch instead?

Reply:
At this late time in 16.3, I'd like to preserve the environment variable Just In Case™. I agree that it Should Be redundant now and at some future time we should remove it.